cocotb/binary.py

```diff
@@ -193,42 +193,56 @@ def assign(self, value):
             )
 
     def _convert_to_unsigned(self, x):
-        x = bin(x)
-        if x[0] == '-':
-            raise ValueError('Attempt to assigned negative number to unsigned '
-                             'BinaryValue')
-        return self._adjust_unsigned(x[2:])
+        if x == 0:
+            return self._adjust_unsigned("")
+        else:
+            x = bin(x)
+            if x[0] == '-':
+                raise ValueError('Attempt to assigned negative number to unsigned '
+                                 'BinaryValue')
+            return self._adjust_unsigned(x[2:])
 
     def _convert_to_signed_mag(self, x):
-        x = bin(x)
-        if x[0] == '-':
-            binstr = self._adjust_signed_mag('1' + x[3:])
+        if x == 0:
+            return self._adjust_unsigned("")
         else:
-            binstr = self._adjust_signed_mag('0' + x[2:])
-        if self.big_endian:
-            binstr = binstr[::-1]
-        return binstr
+            x = bin(x)
+            if x[0] == '-':
+                binstr = self._adjust_signed_mag('1' + x[3:])
+            else:
+                binstr = self._adjust_signed_mag('0' + x[2:])
+            if self.big_endian:
+                binstr = binstr[::-1]
+            return binstr
 
     def _convert_to_twos_comp(self, x):
         if x < 0:
             binstr = bin(2 ** (_clog2(abs(x)) + 1) + x)[2:]
             binstr = self._adjust_twos_comp(binstr)
+        elif x == 0:
+            binstr = ""
```

Comment:
This still needs to use `_adjust_twos_comp` I think?

Reply:
I added the  `_adjust_twos_comp` function. Fixup: ef4e1641845fb8a2a5bdfd5a612394d9e479031f